python/pyarrow/_json.pxd

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# cython: language_level = 3
+
+from pyarrow.includes.libarrow cimport *
+from pyarrow.lib cimport _Weakrefable
+
+
+cdef class ParseOptions(_Weakrefable):
+    cdef:
+        CJSONParseOptions options
+
```

Comment:
Can we structure these more like we have the CSV options?  It's not clear to users what they can set here and doing something like...

```
options = json.ParseOptions()
options.options.xyz = ...
```

...seems a bit redundant.

Reply:
Hi, the CSV options also use this structure. 
https://github.com/apache/arrow/blob/caa78981259fc029182e677f13183371822ce410/python/pyarrow/_csv.pxd#L32-L38
For Arrow developers who use cython, it indeed seems redundant, but for users, the C part is transparent, users only need to functions of Python part to utilize ParseOptions.

Reply:
Ah, sorry, I was referring to something like this: https://github.com/apache/arrow/blob/main/python/pyarrow/_csv.pyx#L89-L205

I don't see an equivalent for JSON (I see there is a `pyx` file with `ParseOptions` but it does not have a docstring or an `__init__` method that allows users to set the various options).

Reply:
I believe there is a  __init__ method of ParseOptions in _json.pyx 

https://github.com/apache/arrow/blob/main/python/pyarrow/_json.pyx#L115-L123